src/paddle.cpp

```diff
@@ -35,32 +36,33 @@ Paddle::~Paddle() {}
 void Paddle::update(float frameTime) {
 	Entity::update(frameTime);
 
-	float yVelocity = 0.0f;
+	if (magnetInitialized)
+		magnetTimer -= frameTime;
 
-	if (input->isKeyDown(controls.up)) {
-		if (getY() > TOP_WALL) {	// paddle is below the top wall
-			yVelocity = -paddleNS::VELOCITY * yVelocityMultipler;
+	if (enableControls) {
+		float yVelocity = 0.0f;
+
+		if (input->isKeyDown(controls.up)) {
+			if (getY() > TOP_WALL) {	// paddle is below the top wall
+				yVelocity = -paddleNS::VELOCITY * yVelocityMultipler;
+			}
 		}
-	}
 
-	if (input->isKeyDown(controls.down)) {
-		if (getY() + (paddleNS::HEIGHT * getScaleY()) < BOTTOM_WALL) {	// paddle is above the bottom wall
-			yVelocity = paddleNS::VELOCITY * yVelocityMultipler;
+		if (input->isKeyDown(controls.down)) {
+			if (getY() + (paddleNS::HEIGHT * getScaleY()) < BOTTOM_WALL) {	// paddle is above the bottom wall
+				yVelocity = paddleNS::VELOCITY * yVelocityMultipler;
+			}
 		}
-	}
 
-	setVelocity(VECTOR2(0, yVelocity));
+		setVelocity(VECTOR2(0, yVelocity));
 
-	if (magnetised && magnetBall != nullptr) {
-		magnetBall->setVelocity(VECTOR2(0, yVelocity));
-	}
+		if (magnetised && magnetBall != nullptr) {
+			magnetBall->setVelocity(VECTOR2(0, yVelocity));
+		}
 
-	if (magnetInitialized) {
-		magnetTimer -= frameTime;
+		spriteData.x += frameTime * velocity.x;
+		spriteData.y += frameTime * velocity.y;
```

Comment:
Have to write in ai() because the new velocities are calculated there. For the AI player, the update() mtd would do nothing except keep track of `magnetTimer`.